### PR TITLE
Add param to delete workflow API to clear status even if resources exist

### LINKED ISSUE #692: [FEATURE] Add a parameter to force deletion of workflow state document when a template is deleted
Coming from https://github.com/opensearch-project/flow-framework/pull/689#issuecomment-2086247313

### Is your feature request related to a problem?

Presently when a template is deleted, the workflow state is not deleted if resources exist.

However, a user may wish to delete the state document anyway, and track the resources separately.

### What solution would you like?

Add a REST path parameter such as "force" or "delete_state" or similar (defaulting to false) to delete the state document when the template is deleted.

### What alternatives have you considered?

A separate API to delete the state document.

A combined API to delete the template and state together.

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Adds an optional `clear_status` param to the delete workflow API (defaulting `false`).

If present, overrides the "do resources exist" check and permits deleting the workflow state as long as provisioning is not `IN_PROGRESS`

Please also review Documentation PR: https://github.com/opensearch-project/documentation-website/pull/7268

### Issues Resolved

Resolves #692

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java [hunk 1]
```diff
@@ -62,6 +63,7 @@ public List<Route> routes() {
     @Override
     protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         String workflowId = request.param(WORKFLOW_ID);
+        request.param(CLEAR_STATUS); // consume and ignore, we will pass params to workflow
```
Comment (joshpalis) [inline src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java:66]: Just to refresh my memory, why do we need to invoke `param()` for each of these? We consume all the params either way when we send a transport request

**Review by joshpalis — APPROVED** [review]
Overall LGTM, just had a question

File: src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java (hunk 1 above)
Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java:66]: First, I had neglected to push a commit I'd saved locally that changes this to actually pull just this param and send just that map over transport.  About to do that now.

But the answer to the larger question is that if we throw an exception _before_ getting to the `params()` call later, such as if the plugin is disabled, we won't consume it and will get a misleading error message.

When I first tried to locally run integ tests before putting this here, I got this:

> > Task :integTest
> 
> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.flowframework.rest.FlowFrameworkRestApiIT.testCreateAndProvisionRemoteModelWorkflow" -Dtests.seed=FBBABD239888A9F8 -Dtests.security.manager=false -Dtests.locale=en-AU -Dtests.timezone=Antarctica/Macquarie -Druntime.java=17
> 
> org.opensearch.flowframework.rest.FlowFrameworkRestApiIT > testCreateAndProvisionRemoteModelWorkflow FAILED
>     org.opensearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:56350], URI [/_plugins/_flow_framework/workflow/qai-xY8B6INOlEYKiSbi?clear_status=true], status line [HTTP/1.1 400 Bad Request]
>     {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"request [/_plugins/_flow_framework/workflow/qai-xY8B6INOlEYKiSbi] contains unrecognized parameter: [clear_status]"}],"type":"illegal_argument_exception","reason":"request [/_plugins/_flow_framework/workflow/qai-xY8B6INOlEYKiSbi] contains unrecognized parameter: [clear_status]"},"status":400}

Comment (owaiskazi19) [reply src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java:66]: Similar to https://github.com/opensearch-project/flow-framework/pull/295?

**Review by owaiskazi19 — APPROVED** [review]
LGTM! Thanks for addressing this issue

File: src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java (hunk 1 above)
Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/rest/RestDeleteWorkflowAction.java:66]: > Similar to #295?

Exactly.

**Pull request merged**